routes/api/branches.js

```diff
@@ -1,8 +1,67 @@
 const express = require("express");
 const router = express.Router();
 
-router.get("/test", (req, res) =>
-  res.json({ msg: "This is the branches route" })
-);
+const Branch = require("../../models/Branch");
+const TextString = require('../../models/TextString');
+
+// Show
+router.get("/:id", (req, res) => {
+  Branch.findById(req.params.id)
+    .then((branch) => {
+      if (branch) {
+        res.json(branch);
+      } else {
+        res.status(404).json("No Such Branch");
+      }
+    })
+    .catch((err) => res.status(404).json(err));
+});
+
+// Index
+router.get("/", (req, res) => {
+  Branch.find({})
+    .then((branch) => {
+      res.json(branch);
+    })
+    .catch((err) => res.status(404).json(err));
+});
+
+// Create
+router.post("/", async (req, res) => {
+  Branch.create(req.body)
+      .then((branch) => res.json(branch))
+      .catch((err) => res.json(err));
+});
+
+// Update
+router.put("/:id", async function (req, res) {
+ Branch.findOneAndUpdate({ _id: req.params.id }, req.body, {
+      new: true,
+      useFindAndModify: false,
+    })
+      .then((branch) => res.json(branch))
+      .catch((err) => res.json(err));
+});
+
+
+// Delete
+router.delete("/:id", function (req, res) {
+    Branch.deleteOne({ _id: req.params.id }, err => {
+        if(err) console.log(err);
```

Comment:
What's the difference between this `err` than the next one?
Do we need this check? If not then remove the condition